iconservice/iconscore/internal_call.py

```diff
@@ -107,61 +118,91 @@ def _make_trace(self,
 
         arg_data = arg_data1 + arg_data2
         trace = Trace(_from, TraceType.CALL, [_to, func_name, arg_data, amount])
-        self.__context.traces.append(trace)
+        context.traces.append(trace)
 
-    def _other_score_call(self,
+    @staticmethod
+    def _other_score_call(context: 'IconScoreContext',
                           addr_from: Address,
                           addr_to: 'Address',
-                          func_name: Optional[str],
-                          arg_params: tuple,
-                          kw_params: dict,
-                          amount: int) -> Any:
+                          func_name: str,
+                          amount: int,
+                          arg_params: Optional[tuple],
+                          kw_params: Optional[dict]) -> Any:
         """Call the functions provided by other icon scores.
 
+        :param context:
         :param addr_from:
         :param addr_to:
         :param func_name:
+        :param amount:
         :param arg_params:
         :param kw_params:
-        :param amount:
+
         :return:
         """
-        IconScoreContextUtil.validate_score_blacklist(self.__context, addr_to)
 
-        if len(self.__context.msg_stack) == MAX_CALL_STACK_SIZE:
+        IconScoreContextUtil.validate_score_blacklist(context, addr_to)
+
+        if len(context.msg_stack) == MAX_CALL_STACK_SIZE:
             raise InvalidRequestException('Max call stack size exceeded')
 
-        self.__context.msg_stack.append(self.__context.msg)
```

Comment:
Not now but I think self.__context.msg can be replaced with self.__context.msg_stack, removing self.__context.msg.

Reply:
if you understand that mean which I want to fix at #20 (comment)
please review again about this.

Reply:
Let's handle the issues about trace on another branch.